syntax = "proto3";

package variable_service;

message WriteRequest {
  string key = 1;
  int64 value = 2;
}
message WriteResponse {
  bool was_inserted = 1;
}

message ReadRequest {
  string key = 1;
}
message ReadResponse {
  int64 value = 1;
}

message DelRequest {
  string key = 1;
}
message DelResponse {
  bool was_deleted = 1;
}

service VariableService {
  rpc Write(WriteRequest) returns(WriteResponse); // The forbidden upsert
  rpc Read(ReadRequest) returns(ReadResponse);
  rpc Del(DelRequest) returns(DelResponse);
}